src/Aspire.Hosting.Keycloak/KeycloakResourceBuilderExtensions.cs

```diff
@@ -56,19 +66,31 @@ public static IResourceBuilder<KeycloakResource> AddKeycloak(
 
         var resource = new KeycloakResource(name, adminUsername?.Resource, passwordParameter);
 
+        var targetPort = port == HttpsContainerPort ? HttpsContainerPort : DefaultContainerPort;
+
         var keycloak = builder
             .AddResource(resource)
             .WithImage(KeycloakContainerImageTags.Image)
             .WithImageRegistry(KeycloakContainerImageTags.Registry)
             .WithImageTag(KeycloakContainerImageTags.Tag)
-            .WithHttpEndpoint(port: port, targetPort: DefaultContainerPort)
+            .WithHttpEndpoint(port: port, targetPort: targetPort)
             .WithHttpEndpoint(targetPort: ManagementInterfaceContainerPort, name: ManagementEndpointName)
             .WithHttpHealthCheck(endpointName: ManagementEndpointName, path: "/health/ready")
             .WithEnvironment(context =>
             {
                 context.EnvironmentVariables[AdminEnvVarName] = resource.AdminReference;
                 context.EnvironmentVariables[AdminPasswordEnvVarName] = resource.AdminPasswordParameter;
                 context.EnvironmentVariables[HealthCheckEnvVarName] = "true";
+                if (port == HttpsContainerPort) {
```

Comment:
I'm not following why this is the predicate for setting all the config to setup Keycloak for running behind a reverse proxy. My reading of this is that when the user specifies the `port` to be `8443`, we configure the Keycloak container as if it's running behind a reverse proxy, but also enable non-HTTPS? How is it expected this will be used, i.e. what will the app host code look like?

Reply:
Ok. I've simplified this quite a bit with recent commit for bare minimum for running behind reverse proxy.

Now, there are only 3 additional env vars:

```
    KC_HTTP_ENABLED=true
    KC_PROXY-HEADERS=xforwarded
    KC_HOSTNAME-STRICT=false
```

Technically, `KC_HOSTNAME-STRICT` doesn't have to be false and shouldn't be for prod, but if it's not, `KC_HOSTNAME` env var also has to be provided and needs to match the ACA container url or a configured custom domain.

Reply:
That makes sense as I did mention in my initial PR description and I thought that might be preferred choice with a dedicated `Publish` method similar to `PublishAsAzurePostgresFlexibleServer`.

So, maybe we wrap this additional config in a new `PublishAsAzureContainerApp` extension method which could set the `KC_HTTP_ENABLED` and `KC_PROXY-HEADERS` env vars and the `KC_HOSTNAME` could be derived from the EndpointReference and or a custom domain, if provided, via param on the new method.

Let me know if that jives with the direction you're thinking.

Reply:
Something like `WithReverseProxy` I think is the basic building block. Folks can then put that behind an `if` check for publish mode. We can consider introducing something like `PublishWithReverseProxy` if we want to make that a wider pattern (it isn't today, but I for one think we should consider it).

Reply:
We make the same default assumption in project resources https://github.com/dotnet/aspire/blob/ea5850afad161c0e850fe0ea07fc27a841bb61b8/src/Aspire.Hosting/ProjectResourceBuilderExtensions.cs#L316-L319

We have a disable flag.

Reply:
@davidfowl fair point. It's likely there are other hosting integrations we could apply this assumption to as well then.